docs/markdown/snippets/java_module.md

```diff
@@ -0,0 +1,9 @@
+## Java Module
+
+The Java module has been added to Meson. The Java module allows users to
+generate native header files without needing to use a `custom_target()`.
+
+```meson
+jmod = import('java')
+native_header = jmod.generate_native_header('File.java', package: 'com.mesonbuild')
```

Comment:
This should also show how to pass `native_header` to a Java target.

Reply:
Please let me know if the update is enough. I'm on my phone and away from keyboard atm.